infer/environment.cc

```diff
@@ -1011,13 +1012,18 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
                     }
 
                     if (it->main.method.exists() && it->main.method.data(ctx)->flags.isPackagePrivate) {
-                        core::ClassOrModuleRef klass = core::Types::getClassForAppliedOrClassType(it->main.receiver);
+                        core::ClassOrModuleRef klass = it->main.method.data(ctx)->owner;
                         if (klass.exists()) {
                             const auto &curPkg = ctx.state.packageDB().getPackageForFile(ctx, ctx.file);
-                            if (curPkg.mangledName() != core::NameRef::noName() && !curPkg.ownsSymbol(ctx, klass)) {
+                            if (curPkg.exists() && !curPkg.ownsSymbol(ctx, klass)) {
                                 if (auto e = ctx.beginError(bind.loc, core::errors::Infer::PackagePrivateMethod)) {
-                                    e.setHeader("Non-package-private call to package-private method `{}` on `{}`",
-                                                it->main.method.data(ctx)->name.show(ctx), it->main.receiver.show(ctx));
+                                    auto &curPkgName = curPkg.fullName();
+                                    // TODO (aadi-stripe, add name of owning package to message).
```

Comment:
I'll be adding this later. Right now going from symbol -> owning package is a bit clunky as we have to manipulate the mangled name, but we can fix that later. For now I feel that this error message is descriptive enough for the Stripe codebase.